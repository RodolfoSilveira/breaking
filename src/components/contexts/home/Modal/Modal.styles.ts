import styled from 'styled-components'
import media, { generateMedia } from 'styled-media-query';

const customMedia = generateMedia({
  fold: '280px'
});

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  &::before{
    content: "";
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100vh;
    background: rgba(0, 0, 0, 0.4);
    z-index: 4;
    filter: blur(2px);
    ${media.lessThan('medium')`
      min-height: 100vh;
    `}
  }
`
export const Container = styled.div`
  top: 60px;
  border-radius: 3px;
  position: absolute;
  background: #fff;
  z-index: 5;
  padding: 40px;
  width: 800px;
  margin: 20px;
  ${media.lessThan('medium')`
    width: 90%;
  `}
  ${customMedia.lessThan('fold')`
    padding: 20px;
  `}
`;

export const Closed = styled.div`
  margin: 20px;
  background: #FFF;
  position: absolute;
  border-radius: 360px;
  top: -80px;
  left: 900px;
  ${media.lessThan('medium')`
    left: 240px;
  `}
  ${media.between('medium', 'large')`
    left: 600px;
  `}
  ${customMedia.lessThan('fold')`
    left: 180px;
  `}
  &:focus {
    border: none;
    outline: none;
  }
`;

export const ButtonClose = styled.button`
  border: none;
  background: none;
  width: 40px;
  height: 40px;
  font-size: 1.2rem;
  font-weight: 500;
  cursor: pointer;
  color: #858585;
  &:focus {
    border: none;
    outline: none;
  }
`;

export const InformationBox = styled.div`
  display: flex;
  ${media.lessThan('medium')`
    flex-wrap: wrap;
  `}
  ${media.between('medium', 'large')`
    flex-wrap: nowrap;
  `}
`;

export const ImageBox = styled.div`
  margin-right: 25px;
  ${media.lessThan('medium')`
    margin: 20px auto;
  `}
  ${media.between('medium', 'large')`
    margin-right: 40px;
  `}
`;

export const Image = styled.img`
  width: 300px;
  height: 100%;
  object-fit: cover;
  ${media.lessThan('medium')`
    width: 200px;
  `}
`;

export const Information = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  height: 520px;
  overflow-y: scroll;
  padding-right: 20px;
  ${media.lessThan('medium')`
    height: auto;
    overflow-y: none;
    padding-right: 0;
  `}
  ${media.between('medium', 'large')`
    height: auto;
    overflow-y: none;
    padding-right: 0;
  `}
  ::-webkit-scrollbar {
    width: 6px;
  }
  ::-webkit-scrollbar-thumb {
    background-color: #4530b7;
    border-radius: 12px;
  }
`;

export const Header = styled.header``;

export const Title = styled.h2`
  font-size: 1.6rem;
  color: #333;
  font-weight: 500;
  margin-bottom: 10px;
  max-lines: 2;
`;

export const Ocuppation = styled.p`
  color:  #4530b7;
  font-size: .8rem;
`;

export const Info = styled.section`
  margin-top: 40px;
`;

export const Span = styled.p`
  color: #333;
  font-weight: normal;
  margin-bottom: 20px;
  font-size: 16px;
`

export const SubTitle = styled.h2`
  color: #333;
  font-weight: normal;
  text-transform: uppercase;
  margin-top: 20px;
  margin-bottom: 20px;
  font-size: 16px;
`;

export const List = styled.ul``;

export const Options = styled.li`
  display: flex;
  justify-content: space-between;
  align-items: center;
  & + & {
    margin-top: 10px;
  }
`;

export const Details = styled.p`
  color: #999;
  font-size: .8rem;
`;

export const Section = styled.p`
  color: #333;
  font-weight: 500;
`;

export const History = styled.section`
  margin-top: 40px;
`;

export const HistoryTitle = styled.h2`
  color: #333;
  font-weight: 500;
  text-transform: uppercase;
  margin-bottom: 20px;
`;

export const Description = styled.p`
  color: #999;
  line-height: 1.6rem;
  > svg {
    width: 26px;
    height: 26px;
    margin-right: 10px;
    position: relative;
    top: 5px;
  }
`;
