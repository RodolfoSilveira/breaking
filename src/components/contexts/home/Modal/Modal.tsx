import { useRef } from 'react'
import { CharacterType } from 'interfaces/characters'
import { useClickOutside } from 'hooks'
import { ErrorBoundary } from 'lib/core/ErrorBoundary'
import * as S from './Modal.styles'

export type ModalProps = {
  character: CharacterType
  closedModal: () => void
}

const Modal = ({ character, closedModal }: ModalProps) => {
  const wrapperRef = useRef(null);
  useClickOutside(wrapperRef, closedModal);

  return (
    <S.Wrapper>
      <S.Container ref={wrapperRef}>
        <S.Closed>
          <S.ButtonClose onClick={closedModal} type="button">X</S.ButtonClose>
        </S.Closed>
        <S.InformationBox>
          <S.ImageBox>
            <S.Image src={character.img} alt="profile" />
          </S.ImageBox>
          <S.Information>
            <S.Header>
              <S.Title>
                {character.name || 'unknown'}
              </S.Title>
              <S.Span>
                {character.nickname || 'unknown'}
              </S.Span>
            </S.Header>
            <S.SubTitle>Ocupation</S.SubTitle>
            <ErrorBoundary>
              <S.List>
                {character.occupation.map((item: string, index: number) => (
                  <S.Options key={String(index)}>
                    <S.Ocuppation>
                      {item}
                    </S.Ocuppation>
                  </S.Options>
                ))}
              </S.List>
            </ErrorBoundary>
            <S.Info>
              <S.SubTitle>Information</S.SubTitle>
              <S.List>
                <S.Options>
                  <S.Section>Birthday</S.Section>
                  <S.Details>{character.birthday || 'unknown'}</S.Details>
                </S.Options>
                <S.Options>
                  <S.Section>Portrayed</S.Section>
                  <S.Details>{character.portrayed || 'unknown'}</S.Details>
                </S.Options>
                <S.Options>
                  <S.Section>Category</S.Section>
                  <S.Details>{character.category || 'unknown'}</S.Details>
                </S.Options>
                <S.Options>
                  <S.Section>Status</S.Section>
                  <S.Details>{character.status || 'unknown'}</S.Details>
                </S.Options>
              </S.List>
            </S.Info>
          </S.Information>
        </S.InformationBox>
      </S.Container>
    </S.Wrapper>
  )
}


export default Modal
