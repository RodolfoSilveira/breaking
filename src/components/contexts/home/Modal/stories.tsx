import { Story, Meta } from '@storybook/react/types-6-0'
import Modal, { ModalProps } from './Modal'

export default {
  title: 'Design System/Modal',
  component: Modal,
  parameters: {
    docs: {
      description: {
        component: 'Modal',
      },
    },

  },
} as Meta

const Template: Story<ModalProps> = (args) => <Modal {...args} />

export const DefaultModal = Template.bind({});

DefaultModal.args = {
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  closedModal: () => { },
  character: {
    name: 'Walter White',
    nickname: 'Heisenberg',
    occupation: ['High School Chemistry Teacher', 'Meth King Pin'],
    portrayed: 'Bryan Cranston',
    status: 'Presumed dead',
    img: 'https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg',
    char_id: 1,
    appearance: [1, 2, 3, 4, 5],
    better_call_saul_appearance: [],
    birthday: '09-07-1958',
    category: 'Breaking Bad'
  }
}
