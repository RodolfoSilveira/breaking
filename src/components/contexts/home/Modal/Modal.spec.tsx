/* eslint-disable @typescript-eslint/no-empty-function */
import { screen, render } from '@testing-library/react';
import Modal from '.'

const character = {
  char_id: 1,
  name: 'teste',
  birthday: '',
  occupation: ['teste'],
  img: 'teste',
  status: 'teste',
  nickname: 'teste',
  appearance: [],
  portrayed: 'teste',
  category: 'teste',
  better_call_saul_appearance: [],
}

describe('<Modal />', () => {
  it('should render a Modal', () => {
    const { container } = render(<Modal
      character={character}
      closedModal={() => {}}
    />)

    expect(screen.getByText(/unknown/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
});
