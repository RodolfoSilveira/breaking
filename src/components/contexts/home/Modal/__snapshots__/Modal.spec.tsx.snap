// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render a Modal 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0::before {
  content: "";
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  background: rgba(0,0,0,0.4);
  z-index: 4;
  -webkit-filter: blur(2px);
  filter: blur(2px);
}

.c1 {
  top: 60px;
  border-radius: 3px;
  position: absolute;
  background: #fff;
  z-index: 5;
  padding: 40px;
  width: 800px;
  margin: 20px;
}

.c2 {
  margin: 20px;
  background: #FFF;
  position: absolute;
  border-radius: 360px;
  top: -80px;
  left: 900px;
}

.c2:focus {
  border: none;
  outline: none;
}

.c3 {
  border: none;
  background: none;
  width: 40px;
  height: 40px;
  font-size: 1.2rem;
  font-weight: 500;
  cursor: pointer;
  color: #858585;
}

.c3:focus {
  border: none;
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  margin-right: 25px;
}

.c6 {
  width: 300px;
  height: 100%;
  object-fit: cover;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 520px;
  overflow-y: scroll;
  padding-right: 20px;
}

.c7::-webkit-scrollbar {
  width: 6px;
}

.c7::-webkit-scrollbar-thumb {
  background-color: #4530b7;
  border-radius: 12px;
}

.c8 {
  font-size: 1.6rem;
  color: #333;
  font-weight: 500;
  margin-bottom: 10px;
  max-lines: 2;
}

.c13 {
  color: #4530b7;
  font-size: .8rem;
}

.c14 {
  margin-top: 40px;
}

.c9 {
  color: #333;
  font-weight: normal;
  margin-bottom: 20px;
  font-size: 16px;
}

.c10 {
  color: #333;
  font-weight: normal;
  text-transform: uppercase;
  margin-top: 20px;
  margin-bottom: 20px;
  font-size: 16px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 + .c11 {
  margin-top: 10px;
}

.c16 {
  color: #999;
  font-size: .8rem;
}

.c15 {
  color: #333;
  font-weight: 500;
}

@media (max-width:768px) {
  .c0::before {
    min-height: 100vh;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 90%;
  }
}

@media (max-width:280px) {
  .c1 {
    padding: 20px;
  }
}

@media (max-width:768px) {
  .c2 {
    left: 240px;
  }
}

@media (min-width:768px) and (max-width:1170px) {
  .c2 {
    left: 600px;
  }
}

@media (max-width:280px) {
  .c2 {
    left: 180px;
  }
}

@media (max-width:768px) {
  .c4 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

@media (min-width:768px) and (max-width:1170px) {
  .c4 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

@media (max-width:768px) {
  .c5 {
    margin: 20px auto;
  }
}

@media (min-width:768px) and (max-width:1170px) {
  .c5 {
    margin-right: 40px;
  }
}

@media (max-width:768px) {
  .c6 {
    width: 200px;
  }
}

@media (max-width:768px) {
  .c7 {
    height: auto;
    overflow-y: none;
    padding-right: 0;
  }
}

@media (min-width:768px) and (max-width:1170px) {
  .c7 {
    height: auto;
    overflow-y: none;
    padding-right: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <button
          class="c3"
          type="button"
        >
          X
        </button>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            alt="profile"
            class="c6"
            src="teste"
          />
        </div>
        <div
          class="c7"
        >
          <header
            class=""
          >
            <h2
              class="c8"
            >
              teste
            </h2>
            <p
              class="c9"
            >
              teste
            </p>
          </header>
          <h2
            class="c10"
          >
            Ocupation
          </h2>
          <ul
            class=""
          >
            <li
              class="c11 c12"
            >
              <p
                class="c13"
              >
                teste
              </p>
            </li>
          </ul>
          <section
            class="c14"
          >
            <h2
              class="c10"
            >
              Information
            </h2>
            <ul
              class=""
            >
              <li
                class="c11 c12"
              >
                <p
                  class="c15"
                >
                  Birthday
                </p>
                <p
                  class="c16"
                >
                  unknown
                </p>
              </li>
              <li
                class="c11 c12"
              >
                <p
                  class="c15"
                >
                  Portrayed
                </p>
                <p
                  class="c16"
                >
                  teste
                </p>
              </li>
              <li
                class="c11 c12"
              >
                <p
                  class="c15"
                >
                  Category
                </p>
                <p
                  class="c16"
                >
                  teste
                </p>
              </li>
              <li
                class="c11 c12"
              >
                <p
                  class="c15"
                >
                  Status
                </p>
                <p
                  class="c16"
                >
                  teste
                </p>
              </li>
            </ul>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
